asm_to_pil/src/lib.rs

```diff
@@ -1,952 +1,127 @@
-                    self.process_assignment_value(*left),
-                    self.process_assignment_value(*right),
-                ),
-                BinaryOperator::Sub => self.add_assignment_value(
-                    self.process_assignment_value(*left),
-                    self.negate_assignment_value(self.process_assignment_value(*right)),
-                ),
-                BinaryOperator::Mul => {
-                    let left = self.process_assignment_value(*left);
-                    let right = self.process_assignment_value(*right);
-                    if let [(f, AffineExpressionComponent::Constant)] = &left[..] {
-                        // TODO overflow?
-                        right
-                            .into_iter()
-                            .map(|(coeff, comp)| (*f * coeff, comp))
-                            .collect()
-                    } else if let [(f, AffineExpressionComponent::Constant)] = &right[..] {
-                        // TODO overflow?
-                        left.into_iter()
-                            .map(|(coeff, comp)| (*f * coeff, comp))
-                            .collect()
-                    } else {
-                        panic!("Multiplication by non-constant.");
-                    }
-                }
-                BinaryOperator::Pow => {
-                    let left = self.process_assignment_value(*left);
-                    let right = self.process_assignment_value(*right);
-                    if let (
-                        [(l, AffineExpressionComponent::Constant)],
-                        [(r, AffineExpressionComponent::Constant)],
-                    ) = (&left[..], &right[..])
-                    {
-                        // TODO overflow?
-                        if r.to_arbitrary_integer() > (u32::MAX).into() {
-                            panic!("Exponent too large");
-                        }
-                        vec![(l.pow(r.to_integer()), AffineExpressionComponent::Constant)]
-                    } else {
-                        panic!("Exponentiation of non-constants.");
-                    }
-                }
-                BinaryOperator::Div
-                | BinaryOperator::Mod
-                | BinaryOperator::BinaryAnd
-                | BinaryOperator::BinaryXor
-                | BinaryOperator::BinaryOr
-                | BinaryOperator::ShiftLeft
-                | BinaryOperator::ShiftRight => {
-                    panic!("Invalid operation in expression {left} {op} {right}")
-                }
-            },
-            Expression::UnaryOperation(op, expr) => {
-                assert!(op == UnaryOperator::Minus);
-                self.negate_assignment_value(self.process_assignment_value(*expr))
+    pub fn parse_instruction<T: FieldElement>(input: &str) -> Instruction<T> {
+        match parser::powdr::InstructionDeclarationParser::new()
+            .parse(&format!("instr dummy {input}"))
```

Comment:
why dummy?

Reply:
I was reusing a parser for the whole instruction to get only the right hand side (and ignoring the name). Refactored to using a parser for the right hand side only.